apps/backend/src/api/monitoring.py

```
@@ -0,0 +1,491 @@
+                "unhealthy_devices": len([d for d in device_health if d["health_score"] < 3])
+            },
+            "timestamp": datetime.now(UTC).isoformat()
+        }
+        
+    except Exception as e:
+        logger.error(f"Polling health check failed: {e}")
+        raise HTTPException(status_code=500, detail=f"Polling health check failed: {str(e)}")
+
+
+@router.get("/performance")
+@limiter.limit("10/minute")  
```

Ruff (0.12.2)

Trailing whitespace

Remove trailing whitespace

(W291)